Ship direction can be explained recursively

\begin{equation}
    \begin{cases}
        \eta_k &= \eta_{k - 1} + \xi_k, \\
        \eta_0 &= 0,
    \end{cases}
    \Rightarrow
    \eta_k = \sum\limits_{i = 1}^k \xi_k
\end{equation}

where $\xi_k \sim N\left( 0, \frac{1}{16} \right)$,
and thus $\eta_k \sim N\left(0, \frac{k}{16} \right)$.

Thus, position of the ship is a process that depends on the $\eta$

\begin{equation}
    \begin{cases}
        \zeta^x_k &= \zeta^x_{k - 1} - 15 \cdot \cos{\eta_k}, \\
        \zeta^y_k &= \zeta^y_{k - 1} + 15 \cdot \sin{\eta_k}, \\
        \zeta^x_0 &= 500, \\
        \zeta^y_0 &= 0,
    \end{cases}
    \Rightarrow
    \begin{cases}
        \zeta^x_k &= 500 - 15 \cdot \sum\limits_{i = 1}^k \cos{\xi_i}, \\
        \zeta^y_k &= 15 \cdot \sum\limits_{i = 1}^k \sin{\xi_i},
    \end{cases}
\end{equation}

The task is to plot distributions of $\pmb{\zeta}_{12}$ and $\pmb{\zeta}_{24}$ vectors.

Assume that $\pmb{\zeta}_k$ is a discrete random vector.
Then, its probability to be equal to a certain vector $\pmb{v}$ is

\begin{equation}
    P\left( \pmb{\zeta}_k = \pmb{v} \right)
    = P\left( \zeta^x_k = v^x, \zeta^y_k = v^y \right)
    = P\left( 500 - 15 \cdot \sum\limits_{i = 1}^k \cos{\xi_i} = v^x, 15 \cdot \sum\limits_{i = 1}^k \sin{\xi_i} = v^y \right)
    = P\left( 33 \frac{1}{3} - \sum\limits_{i = 1}^k \cos{\xi_i} = \frac{v^x}{15}, \sum\limits_{i = 1}^k \sin{\xi_i} = \frac{v^y}{15} \right).
\end{equation}

Let's transform our axes:
- scale them down by factor $15$
- move center of coordinates $33 \frac{1}{3}$ units right
- invert the $x$ axis

Now, we have a new random vector

\begin{equation}
    \begin{cases}
        \theta^x_k &= \sum\limits_{i = 1}^k \cos{\xi_i}, \\
        \theta^y_k &= \sum\limits_{i = 1}^k \sin{\xi_i}.
    \end{cases}
\end{equation}

We need to find a distribution of the random vector $\pmb{\theta}_k$

\begin{equation}
    P\left( \pmb{\theta}_k = \pmb{v} \right)
    = P\left( \left\langle \sum\limits_{i = 1}^k \cos{\xi_i}, \sum\limits_{i = 1}^k \sin{\xi_i} \right\rangle = \left\langle x, y \right\rangle \right)
    = P\left( \sum\limits_{i = 1}^k \left\langle \cos{\xi_i}, \sin{\xi_i} \right\rangle = \left\langle x, y \right\rangle \right)
\end{equation}

In complex form

\begin{equation}
    P\left( \pmb{\theta}_k = \pmb{v} \right)
    = P\left( \sum\limits_{i = 1}^k \left\langle \cos{\xi_i}, \sin{\xi_i} \right\rangle = \left\langle x, y \right\rangle \right)
    = P\left( \sum\limits_{i = 1}^k \exp{j \cdot \xi_i} = x + j \cdot y \right)
\end{equation}

It would be interesting to investigate distribution of this thing (looks like a bivariate lognormal, but has an imaginary part).

It's hard to calculate a distribution of $\zeta^x_k$,
because this will need to calculate a $k$-multiple integral
to have a convolution of probabilities of cosines of dependent normally distributed random variables.
They are dependent, because $\eta_k$ depends on $\eta_i$ through $\xi_i, i < k$.

Change of angle of the ship in $k + 1$st hour doesn't depend on another rotations,
and thus angle $\eta_{k + 1}$ on $k + 1$st step depends only on the angle $\eta_{k + 1}$ on $k$th step
and its change $\xi_{k + 1}$ on $k + 1$st step.

Imagine that we know that the ship is located at $\left\langle x_k, y_k \right\rangle$ and has an angle $\varphi_k$ after $k$ hours trip without satellite connection.
Probability of the ship do be in a point $\left\langle x_{k + 1}, y_{k + 1} \right\rangle$ and be rotated at angle $\varphi_k$ is

\begin{equation}
     P\left( \pmb{\zeta}_{k + 1} = \left\langle x_{k + 1}, y_{k + 1} \right\rangle, \eta_{k + 1} = \varphi_{k + 1}
             \mid \pmb{\zeta}_k = \left\langle x_k, y_k \right\rangle, \eta_k = \varphi_k \right)
     = P\left\{ \pmb{\zeta}_{k + 1} = \left\langle x_k, y_k \right\rangle
              + \left\langle - 15 \cdot \cos{\varphi_{k + 1}}, 15 \cdot \sin{\varphi_{k + 1}} \right\rangle
         \right\},
\end{equation}

Obviously, if $\left( x_k - x_{k + 1} \right)^2 + \left( y_k - y_{k + 1} \right)^2 \neq 15$,
then the probability is $0$, because we cannot reach the destination.
When $x_{k + 1} - x_k \neq 15 \cdot - \cos{\phi_k}$ and $y_{k + 1} - y_k \neq 15 \cdot \sin{\phi_k}$,
the probability is also $0$.

Finally,

\begin{equation}
     P\left( \pmb{\zeta}_{k + 1} = \left\langle x_{k + 1}, y_{k + 1} \right\rangle,
             \eta_{k + 1} = \varphi_{k + 1}
             \mid \pmb{\zeta}_k = \left\langle x_k, y_k \right\rangle, \eta_k = \varphi_k \right)
     = P\left\{ \xi_{k + 1} = \varphi_{k + 1} - \varphi_k \right\}
         \cdot \mathbb{1}\left( \left( x_k - x_{k + 1} \right)^2 + \left( y_k - y_{k + 1} \right)^2 = 15 \right)
         \cdot \mathbb{1}\left( x_{k + 1} - x_k = - 15 \cdot \cos{\phi_k} \right)
         \cdot \mathbb{1}\left( y_{k + 1} - y_k = 15 \cdot \sin{\phi_k} \right).
\end{equation}

It's quiet simple to take a probability of the ship to be at $\left\langle x_{k + 1}, y_{k + 1} \right\rangle$ and be turned at $\varphi_{k + 1}$,
if we know the probability of it to be at $\left\langle x_k, y_k \right\rangle$ and have an angle $\varphi_k$

\begin{equation}
     P\left( \pmb{\zeta}_{k + 1} = \left\langle x_{k + 1}, y_{k + 1} \right\rangle, \eta_{k + 1} = \varphi_{k + 1}, \pmb{\zeta}_k = \left\langle x_k, y_k \right\rangle, \eta_k = \varphi_k \right)
     =
     P\left( \pmb{\zeta}_{k + 1} = \left\langle x_{k + 1}, y_{k + 1} \right\rangle \mid \pmb{\zeta}_k = \left\langle x_k, y_k \right\rangle, \eta_k = \varphi_k \right)
     \cdot P\left( \pmb{\zeta}_k = \left\langle x_k, y_k \right\rangle, \eta_k = \varphi_k \right).
\end{equation}

Now, imagine that we have a probability for each point $\left\langle x^i_k, y^i_k \right\rangle$ and each angle $\varphi^i_k$,
assuming that events under union are independent,
because otherwise it would be too complicated for a test task

\begin{equation}
     P\bigcup_{i \in I} \left( \pmb{\zeta}_{k + 1} = \left\langle x_{k + 1}, y_{k + 1} \right\rangle, \eta_{k + 1} = \varphi_{k + 1},
             \pmb{\zeta}_k = \left\langle x^i_k, y^i_k \right\rangle, \eta_k = \varphi^i_k \right)
     = \sum\limits_{i \in I}
     P\left( \pmb{\zeta}_{k + 1} = \left\langle x^{k + 1}, y^{k + 1} \right\rangle, \eta_{k + 1} = \varphi_{k + 1}
             \mid \pmb{\zeta}_k = \left\langle x^i_k, y^i_k \right\rangle, \eta_k = \varphi^i_k\right)
     \cdot P\left( \pmb{\zeta}_k = \left\langle x^i_k, y^i_k \right\rangle, \eta_k = \varphi^i_k \right)
\end{equation}

We almost have a recurrent formula.
The only missing step is an initial value.
Though, we know it: the probability of the ship to be at its initial point is $1$ according to the task conditions

\begin{equation}
    P\left( \pmb{\zeta}_{k + 1} = \left\langle 500, 0 \right\rangle, \eta_0 = 0 \right)
\end{equation}

In order to take the final density map, we need just to summarize given probabilities by all angles

\begin{equation}
     P\left( \pmb{\zeta}_k = \left\langle x_k, y_k \right\rangle \right)
     = \sum\limits_{i \in I}
         P\left( \pmb{\zeta}_k = \left\langle x_k, y_k \right\rangle, \eta_k = \varphi^i_k \right)
\end{equation}

The idea is to calculate the first "ring" of probabilities (after one hour).
Then, for each point of the ring, calculate "second-order" rings, and so on.
Given a grid with specified density, we can go through each point,
and, for each point of a circle with radius of $15$ nautical miles,
calculate a probability of the ship to be there in one hour,
and multiply it by probability of the ship to be in current point.

Let's investigate how to do this.

Cumulative distribution function of $\sin{\xi_k}$ is

\begin{equation}
    F_{\sin{\xi_k}}\left( x \right) = P\left( \sin{\xi_k} \le x \right)
\end{equation}

Sine is a periodic function, so we cannot just apply $arcsin$ to both sides of the inequality.
Can we?

Let's calculate what we can lose by this trick

\begin{equation}
    P\left( \left| \xi_k \right| \le \pi \right) = 2 \cdot P\left( \xi_k \le -\pi \right) = 2 \cdot F_{\xi_k}\left( -\pi \right)
\end{equation}

In [1]:
from numpy import pi
from scipy.stats import norm

tail_weight = 2 * norm.cdf(-2 * pi, scale=1/4)
print(f"Tail weight is {tail_weight * 100:e}%")

Tail weight is 2.182448e-137%


We can see that if we assume that after one hour angle may not change for more than $\pi$ in any direction,
we don't lose anything,
because the probability $\sim 10^{-137}$ is very small.

Though, we want to sum several random variables under this assumption.
Let $\tilde{p}$ be the probability of an angle to change for more than $\pi$ after an hour in any direction.
Then, the probability of staying in this bound is $1 - \tilde{p}$.
The probability that neither change will overflow the bound within $k$ hours is $\left( 1 - \tilde{p} \right)^k$.
Let's apply binomial theorem to calculate a probability that at least one direction change will overflow specified boundary

\begin{equation}
    1 - \left( 1 - \tilde{p} \right)^n
    = 1 - \sum\limits_{k=0}^n {n \choose k} \cdot \left( -\tilde{p} \right)^k
\end{equation}

In [2]:
from numpy import pi
from scipy.stats import norm
from scipy.special import comb

final_tail = sum(comb(24, i) * (-2 * norm.cdf(-pi, scale=1/4)) ** i for i in range(1, 25))
print(f"Weight of tail after summation of chain of 24 random variables is {-final_tail * 100:e}%")

Weight of tail after summation of chain of 24 random variables is 7.757390e-33%


The tail grows fast, but it still very tiny and we can ignore it.